### PR TITLE
Guard -D_LIBUNWIND_SANDBOX_OTYPES behind an aarch64 check

### PR DESCRIPTION
**Pull request by dstolfa**
With recent changes in https://github.com/CTSRD-CHERI/llvm-project/pull/731, we need to guard these libunwind defines behind aarch64 to avoid breaking other archs (like CHERI-RISC-V).

### CODE REVIEW

File: lib/libgcc_s/Makefile [hunk 1]
```diff
@@ -54,6 +54,10 @@ SRCS+=		s_scalbnl.c
 .endif
 
 SYMBOL_MAPS+=	${.CURDIR}/Symbol-c18n.map
```
Comment (jrtc27) [inline lib/libgcc_s/Makefile:56]: This is wrong for non-c18n?

Comment (dstolfa) [reply lib/libgcc_s/Makefile:56]: What would the best way to check for this be? The rtld Makefile seems to check for purecap + aarch64. Is that sufficient?

Comment (jrtc27) [reply lib/libgcc_s/Makefile:56]: That sounds about right

**Review by brooksdavis — APPROVED** [review]
There's an argument for making the libexec/rtld-elf/Makefile style change in a separate commit, but this should work.

**Comment by dstolfa** [comment]
> There's an argument for making the libexec/rtld-elf/Makefile style change in a separate commit, but this should work.

It wasn't actually my intention to make changes to the rtld Makefile, it was an accident since I was just copying over what rtld was doing to check for c18n, but might as well :). I broke it out into a separate commit.

**Pull request merged**